src/Host/Broker/Impl/NativeMethods.cs

```diff
@@ -232,5 +232,8 @@ internal static unsafe class NativeMethods {
             IntPtr hToken,
             StringBuilder pszProfilePath,
             ref uint cchProfilePath);
+
+        [DllImport("kernel32.dll")]
+        public static extern uint GetLastError();
```

Comment:
This will not work reliably with P/Invoke (because it may invoke other Win32 APIs that do `SetLastError` as part of marshaling). You need to ensure that all functions have `[... SetLastError=true ...]`, and then use `Marshal.GetLastWin32Error`.